vms/platformvm/node_signature.go

```diff
@@ -0,0 +1,145 @@
+package platformvm
+
+import (
+	cryptoc "crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	"github.com/chain4travel/caminogo/ids"
+	"github.com/chain4travel/caminogo/utils/formatting"
+	"github.com/chain4travel/caminogo/utils/hashing"
+	"github.com/chain4travel/caminogo/vms/components/verify"
+)
+
+var (
+	errCertificateDontMatch = errors.New("node certificate key doesn't match node ID")
+	errKeyPairDontMatch     = errors.New("public key doesn't match private key")
+	errWrongCredentialType  = errors.New("wrong credential type")
+	errWrongPublicKeyType   = errors.New("wrong public key type")
+	errWrongPrivateKeyType  = errors.New("wrong private key type")
+	errMissingPEMPK         = errors.New("pem encoded bytes doesn't contain PRIVATE KEY pem block")
+	errMissingPEMCert       = errors.New("pem encoded bytes doesn't contain CERTIFICATE pem block")
+	errNilCredential        = errors.New("nil credential")
+)
+
+const (
+	rsaSignatureLen    = 512
+	x509CertificateLen = 1185 // TODO@ confirm
+)
+
+type RSACredential struct {
```

Comment:
Are we just doing RSA? Is it possible to boot a node with e.g. a EC25519?
More general is there a limit for the length of RSA keys? Theoretically no cuz there are infinite primes, but are there practical limitations?

Reply:
Yea, if I got it right, it depends on key size

Reply:
Node has hardcoded RSA key len and will use it for generating new key-cert pair if no key is present. But it won't do it if this key already exist (as flag or file), so it could get rsa key that was generated outside of node, which could have different length.